xline/src/server/command.rs

```diff
@@ -708,4 +731,98 @@ mod test {
         assert!(lease_leases_cmd.is_conflict(&cmd5)); // lease read and write
         assert!(cmd6.is_conflict(&lease_leases_cmd)); // lease read and write
     }
+
+    #[test]
+    fn test_compaction_txn_conflict() {
+        let compaction_cmd_1 = Command::new(
+            vec![],
+            RequestWithToken::new(RequestWrapper::CompactionRequest(CompactionRequest {
+                revision: 3,
+                physical: false,
+            })),
+            ProposeId::new("id11".to_owned()),
+        );
+
+        let compaction_cmd_2 = Command::new(
+            vec![],
+            RequestWithToken::new(RequestWrapper::CompactionRequest(CompactionRequest {
+                revision: 5,
+                physical: false,
+            })),
+            ProposeId::new("id12".to_owned()),
+        );
+
+        let txn_with_lease_id_cmd = Command::new(
+            vec![KeyRange::new_one_key("key")],
+            RequestWithToken::new(RequestWrapper::TxnRequest(TxnRequest {
+                compare: vec![],
+                success: vec![RequestOp {
+                    request: Some(Request::RequestPut(PutRequest {
+                        key: b"key".to_vec(),
+                        value: b"value".to_vec(),
+                        lease: 123,
+                        ..Default::default()
+                    })),
+                }],
+                failure: vec![],
+            })),
+            ProposeId::new("id6".to_owned()),
+        );
+
+        let txn_cmd_1 = Command::new(
```

Comment:
Make a function/closure to generate txn commands

Reply:
Done